Use ArmaEvents API for event display

NeutraleNull opened the pull request:
This PR updates the frontend part of the HP to use the arma events instead of the our old forum.

I didn't touch any old stuff and mocked the output of AE to resemble the same format as before so the old code can work with it "out of the box".

DerZade reviewed (changes requested):
Thanks a lot for this.

You adjusted the frontend to access arma.events. 
Instead we should keep the frontend implementation and just update the HP API to access arma.events instead of the forum. 

See here:
https://github.com/gruppe-adler/gruppe-adler.de/blob/a3ed07d8a663215828dd67c62663465dfeb49fdf/api/src/utils/EventsService.ts#L51-L89

This ensures that we don’t send unnecessary requests to arma.events and therefore improves performance for the user :)

NeutraleNull commented:
> Thanks a lot for this.
> 
> You adjusted the frontend to access arma.events. Instead we should keep the frontend implementation and just update the HP API to access arma.events instead of the forum.
> 
> See here:
> 
> https://github.com/gruppe-adler/gruppe-adler.de/blob/a3ed07d8a663215828dd67c62663465dfeb49fdf/api/src/utils/EventsService.ts#L51-L89
> 
> This ensures that we don’t send unnecessary requests to arma.events and therefore improves performance for the user :)

I have reverted the changes to the frontend code i made and moved the query logic into the api backend part as recommended.
@DerZade could you test it out for me? I cannot build the backend code locally as i get this:
<img width="617" alt="image" src="https://github.com/user-attachments/assets/d5309cf5-b5e9-4869-b0fc-b9adc7022fbf" />

DerZade committed:
feat: :sparkles: load events from arma.events

Co-authored-by: NeutraleNull <oliverkrue@googlemail.com>

DerZade commented:
@NeutraleNull I took the liberty and force-pushed to your branch. Are you happy with the changes?

NeutraleNull commented:
@DerZade i had a look and build it locally and it looks fine, however i still was unable to do a local serve for testing. From my side it's fine.
